apps/mark/backend/src/app.ts

```diff
@@ -54,7 +60,18 @@ function buildApi(auth: InsertedSmartCardAuthApi) {
         return err(new Error('User is not an election manager'));
       }
 
-      return await auth.readCardDataAsString({ electionHash });
+      const result = await auth.readCardDataAsString({ electionHash });
+      const electionData = result.ok();
+      const electionDefinition = electionData
+        ? safeParseElectionDefinition(electionData).ok()
+        : undefined;
+
+      if (!electionDefinition) {
+        // While we could provide more specific error messages for different error cases, the
+        // frontend doesn't need that much detail
+        return err(new Error('Unable to read election definition from card'));
+      }
+      return ok(electionDefinition);
```

Comment:
Are we using the error case at all? If we're just checking on the frontend that either `electionDefinition` is ok or not rather than using the error message, we could just return `ElectionDefinition | undefined`

Reply:
This comment got me thinking about error handling holistically again since I do ultimately want to introduce a consistent pattern for auth checks via an auth decorator. And thus far, I've been returning error `Result` objects, e.g.

```ts
if (authStatus.status !== 'logged_in') {
  return err(new Error('User is not logged in'));
}
```

This comment is making me realize that requiring every API endpoint to return a `Result` object could become cumbersome for the UI.

An alternative would be to throw a special authorization error on the server that's converted to a 401, which the react-query query client could then globally special case and not trigger the error boundary for.

I've added an item to my miscellaneous cleanup issue to revisit this and am going to leave this specific case as is for now, still using a `Result`.